components/dataflow/predict/component.yaml

```diff
@@ -0,0 +1,31 @@
+name: Predict using TF on Dataflow
+description: |
+  Runs TensorFlow prediction on Google Cloud Dataflow
+  Input and output data is in GCS
+inputs:
+  - {name: Data file pattern,   type: {GCSPath: {data_type: CSV}},        description: 'GCS or local path of test file patterns.'}
```

Comment:
This input is a URI, but I also want to specify the data type so that the user knows the format of data to pass (and so that the type checking works better).
I think I saw the GCPPath class that had the `path_type` and `data_type` properties.

I've recently talked with Ajay and it seemed to me that he was fine with specifying the inner type of the data in addition to the outer type (URI or GCPPath).

Reply:
I said I'd think about it, not that I'm ok with it :-) This certainly requires more thought and design. It would also be best if this aligns with Ning's type definitions as well. I'm not convinced that we need a type called GCSPath either. And the notation here is opaque, as is the the field 'data_type'. There is no accepted schema yet.

Reply:
Could we drop the properties for now?

Reply:
>It would also be best if this aligns with Ning's type definitions as well.

It did align with Ning's type definitions. It's in our released SDK: https://github.com/kubeflow/pipelines/blob/0.1.12/sdk/python/kfp/dsl/_types.py#L62

		Args
		:param path_type: describes the paths, for example, bucket, directory, file, etc
		:param file_type: describes the files, for example, JSON, CSV, etc.

Someone might have made some breaking changes, but there needs to be some limit on the amount of time the PR author needs to chase an ever changing API.

Reply:
>Could we drop the properties for now

OK. I can replace them with strings. I do not want to lose the type checking.

Would that be enough for you?

Reply:
Done. I've removed all properties from `GCSPath` types, adapting the component definitions to the breaking change in v0.1.13.